app/src/main/res/xml/preferences.xml

```diff
@@ -71,11 +51,43 @@
           android:summary="@string/display_campaigns_explanation"
           android:title="@string/display_campaigns" />
 
+    </PreferenceCategory>
+
+    <PreferenceCategory
+        android:title="Uploads">
+
+        <SwitchPreference
+
+          android:defaultValue="true"
+          android:key="useExternalStorage"
+          app:singleLineTitle="false"
+          android:summary="@string/use_external_storage_summary"
+          android:title="@string/use_external_storage" />
+
+        <SwitchPreference
+          android:key="useAuthorName"
+          app:singleLineTitle="false"
+          android:summary="@string/preference_author_name_toggle_summary"
+          android:title="@string/preference_author_name_toggle" />
+
+        <EditTextPreference
+          android:key="authorName"
+          app:singleLineTitle="false"
+          app:dependency="useAuthorName"
+          app:useSimpleSummaryProvider="true"
+          android:title="@string/preference_author_name" />
+
         <SwitchPreference
             android:defaultValue="true"
             android:key="openDocumentPhotoPickerPref"
             android:summary="@string/open_document_photo_picker_explanation"
             android:title="@string/open_document_photo_picker_title"/>
+
+        <SwitchPreference
+          android:key="inAppCameraLocationPref"
+          android:title="@string/in_app_camera_location_permission_title"
+          android:summary="@string/in_app_camera_location_switch_pref_summary"/>
+
```

Comment:
We need to ensure that these two preferences aren't enabled for users who skipped login. For this the code in SettingsFragment needs to be tweaked. Link to corresponding part of the code: https://github.com/commons-app/apps-android-commons/blob/master/app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java#L167-L175

Also, these could be moved above the customer author name settings such that they appear below "Save In-app shots" so that in-app shot related settings appear close together.